src/Data/Functor/Alt.hs

```diff
@@ -153,6 +155,16 @@ class Functor f => Alt f where
 optional :: (Alt f, Applicative f) => f a -> f (Maybe a)
 optional v = Just <$> v <!> pure Nothing
 
+-- | Generic ('<!>'). Caveats:
+--
+--   1. Will not compile if @f@ is a sum type.
+--   2. Will not compile if @f@ is a recursive type.
```

Comment:
I don't understand where this limitation arises. In fact, I am able to use `galt` on this recursive data type:

```hs
data Foo a = MkFoo (Maybe a) (Foo a)
  deriving (Functor, Generic1)

instance Alt Foo
  where (<!>) = galt
```

And similarly for other methods that have a disclaimer about recursive types.

Reply:
I think that's because I misunderstood the implications of the lack of a `:.:` instance. I think it can't be recursive inside another type constructor, but I'm not sure of the best way to express that. Do you have any suggestions? I'll have a think.

Reply:
`(:.:)` has nothing to do with recursive types, it just represents one type constructor applied to another type constructor, the latter of which having an occurrence of the last type variable. For example, this data type:

```hs
data Baz a = MkBaz (Maybe [a])
  deriving (Functor, Generic1)

instance Alt Baz where
  (<!>) = galt
```

Would have a `Rep1` instance that looks roughly like `type Rep1 Baz = Maybe :.: Rec1 []`. And indeed, this `Alt` instance fails to compile:

```
Foo.hs:11:11: error:
    * No instance for (Alt (Maybe :.: Rec1 []))
        arising from a use of `galt'
    * In the expression: galt
      In an equation for `<!>': (<!>) = galt
      In the instance declaration for `Alt Baz'
   |
11 |   (<!>) = galt
   |           ^^^^
```

That being said, I think we can avoid having to explain this subtlety by just giving `(:.:)` an `Alt` instance. We already define an `Alt` instance for `Compose`, after all:

https://github.com/ekmett/semigroupoids/blob/bcf6c0f2ec29c688417f316c270fdf9ea2df866d/src/Data/Functor/Alt.hs#L315-L316

It should be straightforward to adapt this code for `(:.:)`.